pkg/controller/hostpathprovisioner/prometheus.go

```diff
@@ -181,167 +185,6 @@ func (r *ReconcileHostPathProvisioner) deletePrometheusResources(namespace strin
-			},
-		),
-		generateAlertRule(
-			"HPPNotReady",
-			"kubevirt_hpp_cr_ready == 0",
-			"5m",
-			map[string]string{
-				"summary":     "Hostpath Provisioner is not available to use",
-				"runbook_url": fmt.Sprintf(runbookURLTemplate, "HPPNotReady"),
-			},
-			map[string]string{
-				severityAlertLabelKey:     "warning",
-				healthImpactAlertLabelKey: "critical",
-				partOfAlertLabelKey:       partOfAlertLabelValue,
-				componentAlertLabelKey:    componentAlertLabelValue,
-			},
-		),
-		generateAlertRule(
-			"HPPSharingPoolPathWithOS",
-			"kubevirt_hpp_pool_path_shared_with_os == 1",
-			"1m",
-			map[string]string{
-				"summary":     "HPP pool path sharing a filesystem with OS, fix to prevent HPP PVs from causing disk pressure and affecting node operation",
-				"runbook_url": fmt.Sprintf(runbookURLTemplate, "HPPSharingPoolPathWithOS"),
-			},
-			map[string]string{
-				severityAlertLabelKey:     "warning",
-				healthImpactAlertLabelKey: "warning",
-				partOfAlertLabelKey:       partOfAlertLabelValue,
-				componentAlertLabelKey:    componentAlertLabelValue,
-			},
-		),
-	}
-}
-
-func createPrometheusRule(namespace string) *promv1.PrometheusRule {
-	labels := getRecommendedLabels()
-	labels[PrometheusLabelKey] = PrometheusLabelValue
-
-	runbookURLTemplate := getRunbookURLTemplate()
-
-	return &promv1.PrometheusRule{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: promv1.SchemeGroupVersion.String(),
-			Kind:       "PrometheusRule",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      ruleName,
-			Namespace: namespace,
-			Labels:    labels,
-		},
-		Spec: promv1.PrometheusRuleSpec{
-			Groups: []promv1.RuleGroup{
-				{
-					Name:  "hpp.rules",
-					Rules: append(getRecordRules(namespace), getAlertRules(runbookURLTemplate)...),
-				},
-			},
-		},
-	}
-}
-
-func createPrometheusRole(namespace string) *rbacv1.Role {
-	labels := getRecommendedLabels()
-	labels[PrometheusLabelKey] = PrometheusLabelValue
-
-	return &rbacv1.Role{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      rbacName,
-			Namespace: namespace,
-			Labels:    labels,
-		},
-		Rules: []rbacv1.PolicyRule{
-			{
-				APIGroups: []string{
-					"",
-				},
-				Resources: []string{
-					"services",
-					"endpoints",
-					"pods",
-				},
-				Verbs: []string{
-					"get", "list", "watch",
-				},
-			},
-		},
-	}
-}
-
-func createPrometheusRoleBinding(namespace string) *rbacv1.RoleBinding {
-	monitoringNamespace := getMonitoringNamespace()
-	labels := getRecommendedLabels()
-	labels[PrometheusLabelKey] = PrometheusLabelValue
-
-	return &rbacv1.RoleBinding{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      rbacName,
-			Namespace: namespace,
-			Labels:    labels,
-		},
-		RoleRef: rbacv1.RoleRef{
-			APIGroup: "rbac.authorization.k8s.io",
-			Kind:     "Role",
-			Name:     rbacName,
-		},
-		Subjects: []rbacv1.Subject{
-			{
-				Kind:      "ServiceAccount",
-				Namespace: monitoringNamespace,
-				Name:      "prometheus-k8s",
-			},
```

Comment:
@machadovilaca I tried to replace this 2 functions with https://github.com/kubevirt/hostpath-provisioner-operator/pull/417/files#diff-03994b519dcc3183cc444b854d90597e34c2ded183fa56a4a2e22e06409b547bR77